js/widgets/modewidget.js

```diff
@@ -9,152 +9,145 @@
                     '" height="' +
-                    ICONSIZE +
+                    this.CONSTANTS.ICONSIZE +
                     '" width="' +
-                    ICONSIZE +
+                    this.CONSTANTS.ICONSIZE +
                     '" vertical-align="middle">&nbsp;&nbsp;';
             } else {
-                that._playing = true;
+                this._playing = true;
 
-                this.innerHTML =
+                this._playButton.innerHTML =
                     '&nbsp;&nbsp;<img src="header-icons/stop-button.svg" title="' +
                     _("Stop") +
                     '" alt="' +
                     _("Stop") +
                     '" height="' +
-                    ICONSIZE +
+                    this.CONSTANTS.ICONSIZE +
                     '" width="' +
-                    ICONSIZE +
+                    this.CONSTANTS.ICONSIZE +
                     '" vertical-align="middle">&nbsp;&nbsp;';
 
-                that._playAll();
+                this._playAll();
             }
         };
 
-        widgetWindow.addButton(
+        this.widgetWindow.addButton(
             "export-chunk.svg",
-            ICONSIZE,
+            this.CONSTANTS.ICONSIZE,
             _("Save")
-        ).onclick = function() {
-            that._save();
+        ).onclick = ()=>{
+            this._save();
         };
 
-        widgetWindow.addButton(
+        this.widgetWindow.addButton(
             "erase-button.svg",
-            ICONSIZE,
+            this.CONSTANTS.ICONSIZE,
             _("Clear")
-        ).onclick = function() {
-            that._clear();
+        ).onclick = ()=>{
+            this._clear();
         };
 
-        widgetWindow.addButton(
+        this.widgetWindow.addButton(
             "rotate-left.svg",
-            ICONSIZE,
+            this.CONSTANTS.ICONSIZE,
             _("Rotate counter clockwise")
-        ).onclick = function() {
-            that._rotateLeft();
+        ).onclick = ()=>{
+            this._rotateLeft();
         };
 
-        widgetWindow.addButton(
+        this.widgetWindow.addButton(
             "rotate-right.svg",
-            ICONSIZE,
+            this.CONSTANTS.ICONSIZE,
             _("Rotate clockwise")
-        ).onclick = function() {
-            that._rotateRight();
+        ).onclick = ()=>{
+            this._rotateRight();
         };
 
-        widgetWindow.addButton(
+        this.widgetWindow.addButton(
             "invert.svg",
-            ICONSIZE,
+            this.CONSTANTS.ICONSIZE,
             _("Invert")
-        ).onclick = function() {
-            that._invert();
+        ).onclick = ()=>{
+            this._invert();
         };
 
-        widgetWindow.addButton(
+        this.widgetWindow.addButton(
             "restore-button.svg",
-            ICONSIZE,
+            this.CONSTANTS.ICONSIZE,
             _("Undo")
-        ).onclick = function() {
-            that._undo();
+        ).onclick = ()=>{
```

Comment:
We agreed on a 100 character limit per line, so might straighten this one and the ones alike.